router/server.go

```diff
@@ -229,20 +229,33 @@ func (s *Server) ServeHTTP(isSecure bool, w http.ResponseWriter, req *http.Reque
 		return
 	}
 
-	// This redirects all HTTP request to use HTTPS for all connections which originate from a domain name.
-	// We don't redirect IP-address or localhost URLs
+	// Redirect HTTP requests to HTTPS
+	// Requests from IP addressses and localhost are left untouched
 	if s.configHttp.RedirectHTTP && !isSecure && net.ParseIP(req.Host) == nil && req.Host != "localhost" {
-		host := strings.Split(req.Host, ":")[0] // remove port from host, this is safe even when no port is specified
-		target := fmt.Sprintf("https://%s%s", host, req.URL.Path)
-		if s.configHttp.HTTPSPort != 0 {
-			target = fmt.Sprintf("https://%s:%d%s", host, s.configHttp.HTTPSPort, req.URL.Path) // override default HTTPS port
+
+		// Give 404 for appcache manifest request when HTTP redirection is enabled, this clears out old manifest
+		if req.RequestURI == "/manifest.appcache" {
+			http.Error(w, "", http.StatusNotFound)
+			s.errorLog.Info("Appcache manifest cleared")
```

Comment:
This would happen very few times (i.e. probably only once per user ever), so think it might be worthwhile when debugging redirection, or seeing if redirection happens too often